### DISCUSSION TITLE
Wizard: get clearOnUnmount answers back when navigating through form

### DISCUSSION THREAD

**Discussion by CountSigFigs** [data-driven-forms/react-forms#1108] [Need help] [answered]
I am building a wizard form and have conditional questions that should be clearedOnUnmount if the user chooses a different answer before they move t a different page, but should be persisted otherwise when navigating through the form. However, when the user goes to the next page, because the answer is marked, clearOnUnmount, the answer does not persist and if you click on the back button, the answer goes away. How can I keep the answer persisted when the user clicks the next and back button, while still maintaining the clearOnUnmount if the user chooses a different answer? I have a very simple example here:

[ https://stackblitz.com/edit/fdmahc?file=schema.js](url)

So basically if the user clicks on "airforce" a conditional question appears that should clear when the user selects a different option on that page, but should otherwise persist its answer when you click on the next/back button. If you answer that conditional answer then move forward and then back, the answer clears.

**Comment by CountSigFigs** [comment, marked as answer]
Correct link here: https://stackblitz.com/edit/fdmahc?file=schema.js

**Reply by rvsia** [reply]
Hello, you don’t have to use clearOnUnmount in this use case. Our wizards collect only visible fields when going to the next step, so when the second fields is hidden, the value won’t be shown in the submit function.

If you want to clear the selection when users enable the conditional field, you can use a [value listener](https://data-driven-forms.org/examples/value-listener) that will clear the selection if user selects “airforce”. Unfortunately, we can’t use initializeOnMount or set condition, because of the wizard.

**Reply by rvsia** [reply]
You actually can use the condition set, just set the value to undefined in the else branch of the condition. Check this [example](https://data-driven-forms.org/schema/complex-condition-example).

**Reply by Hyperkid123** [reply]
@CountSigFigs as @rvsia pointed out, the wizard is a bit special in the way it actually chooses the submit values. Normally, you would have to use `clearedOnUnmount` prop to delete any "non-visible" values from the form state. However, because the wizard is basically multiple different forms, we are not using the full state manager values, but we only use store and submit "visible" values when switching between steps. We can't really use the `clearedOnUnmount`, because when you switch to the next page, all previous fields do unmount and trigger the unmount event. We would have keep all the fields in DOM which would have a negative impact on performance.

Your example works if you remove the `clearedOnUnmount`. Check the console output when you submit the form.

**Reply by CountSigFigs** [reply]
Thank you both so much. This whole time, I was under the impression that the clearOnUnmount key was needed, but apparently this is not the case. When I got rid of it, the expected behavior occured.
